src/Sylius/Bundle/CoreBundle/Form/Extension/PaymentMethodTypeExtension.php

```diff
@@ -26,12 +27,18 @@
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $gatewayFactory = $options['data']->getGatewayConfig()->getFactoryName();
```

Comment:
Can we make it a responsibility of `GatewayConfigType`? We should have `$options['data']->getGatewayConfig()` as `$options['data']` there.